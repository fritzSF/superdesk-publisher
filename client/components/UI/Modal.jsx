import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

const Modal = ({ isOpen, children }) => {
  return (
    <React.Fragment>
      <div
        className={classNames("modal modal--double", { in: isOpen })}
        data-backdrop="static"
        style={isOpen ? { zIndex: "1050", display: "block" } : null}
      >
        <div className="modal__dialog">
          <div className="modal__content">{children}</div>
        </div>
      </div>
      {isOpen && (
        <div
          className="modal__backdrop fade in modal--double"
          style={{ zIndex: "1049" }}
        />
      )}
    </React.Fragment>
  );
};

Modal.propTypes = {
  isOpen: PropTypes.bool.isRequired
};

export default Modal;
