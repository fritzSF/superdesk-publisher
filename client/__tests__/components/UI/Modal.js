import React from "react";
import Modal from "../../../components/UI/Modal";
import { render, fireEvent, getByText } from "@testing-library/react";

describe("UI/Modal", () => {
  it("renders itself, children correcty", () => {
    const { container, getByText } = render(
      <Modal isOpen={true}>
        <span>child</span>
      </Modal>
    );

    expect(getByText("child")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
