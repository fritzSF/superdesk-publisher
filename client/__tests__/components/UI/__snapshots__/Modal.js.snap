// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Modal renders itself, children correcty 1`] = `
<div
  class="modal modal--double in"
  data-backdrop="static"
  style="z-index: 1050; display: block;"
>
  <div
    class="modal__dialog"
  >
    <div
      class="modal__content"
    >
      <span>
        child
      </span>
    </div>
  </div>
</div>
`;
